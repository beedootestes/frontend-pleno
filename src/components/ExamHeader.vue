<template>
  <section class="questions-header">
    <div class="title-container">
      <ArrowLeftIcon @click="$router.back()" class="icon-back" :size="50" />
      <p>{{ titleExam }}</p>
    </div>

    <div class="beetcoins">
      <span>$</span>
      {{ beetcoins }} / {{ beetcoinsAchieved }}
    </div>

    <div class="more">
      <span>BeeFree - Orgulho LGBTQIA+</span>
      <span>Conhe√ßa mais do beedoo</span>
    </div>
  </section>
</template>

<script setup lang="ts">
  import ArrowLeftIcon from 'vue-material-design-icons/ArrowLeft.vue';

  defineProps({
    titleExam: {
      type: String,
      required: true
    },

    beetcoins: {
      type: Number,
      required: true
    },

    beetcoinsAchieved: {
      type: Number,
      required: true
    }
  })

</script>

<style scoped lang="scss">
  .questions-header {
    display: flex;
    flex-direction: column;
    border-bottom: 1px solid $color-border;
    box-sizing: border-box;
    padding: 15px 0;

    .title-container {
      display: flex;
      align-items: center;
      margin-bottom: 10px;

      .icon-back {
        cursor: pointer;
      }

      p {
        margin: auto;
      }
    }

    .beetcoins {
      width: fit-content;
      border: 1px solid $color-border;
      border-radius: 15px;
      box-sizing: border-box;
      padding: 5px 15px;
      margin-bottom: 20px;

      span {
        margin-right: 15px;
        color: $color-detail;
      }
    }

    .more {
      display: flex;
      flex-direction: row;
      justify-content: space-between;

      span {
        width: fit-content;
        border: 1px solid $color-border;
        padding: 5px;
        border-radius: 15px;
        box-sizing: border-box;
      }
    }
  }
</style>