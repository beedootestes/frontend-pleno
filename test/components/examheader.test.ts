import {describe, expect, test} from "vitest";
import ExamHeader from "../../src/components/ExamHeader.vue";
import {mount, shallowMount, VueWrapper} from "@vue/test-utils";


describe('Test ExamHeader,vue', () => {
  test('mount component', async () => {
    expect(ExamHeader).toBeTruthy()

    const wrapper: VueWrapper<ExamHeader> = shallowMount(ExamHeader, {
      props: {
        titleExam: 'Exam Question',
        beetcoins: 50,
        beetcoinsAchieved: 30
      }
    })

    expect(wrapper.html()).toMatchSnapshot()

    const title = wrapper.find('.title-container')
    const more = wrapper.find('.more')
    const beetcoins = wrapper.find('.beetcoins')

    expect(title.text()).toStrictEqual('Exam Question')
    expect(more.text()).toStrictEqual('BeeFree - Orgulho LGBTQIA+Conhe√ßa mais do beedoo')
    expect(beetcoins.text()).toStrictEqual('$ 50 / 30')
  })
})