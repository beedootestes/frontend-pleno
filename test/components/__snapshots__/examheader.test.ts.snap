// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test ExamHeader,vue > mount component 1`] = `
"<section data-v-667f190b=\\"\\" class=\\"questions-header\\">
  <div data-v-667f190b=\\"\\" class=\\"title-container\\">
    <arrow-left-icon-stub data-v-667f190b=\\"\\" fillcolor=\\"currentColor\\" size=\\"50\\" class=\\"icon-back\\"></arrow-left-icon-stub>
    <p data-v-667f190b=\\"\\">Exam Question</p>
  </div>
  <div data-v-667f190b=\\"\\" class=\\"beetcoins\\"><span data-v-667f190b=\\"\\">$</span> 50 / 30</div>
  <div data-v-667f190b=\\"\\" class=\\"more\\"><span data-v-667f190b=\\"\\">BeeFree - Orgulho LGBTQIA+</span><span data-v-667f190b=\\"\\">Conhe√ßa mais do beedoo</span></div>
</section>"
`;
